azure-pipelines-pr-validation.yml

```
@@ -42,238 +46,296 @@ variables:
+            dropFolder: 'artifacts\VSSetup\$(BuildConfiguration)\Insertion'
+            dropName: $(VisualStudio.DropName)
+            accessToken: $(_DevDivDropAccessToken)
+
+          # Publish insertion packages to CoreXT store.
+          - output: nuget
+            displayName: 'Publish CoreXT Packages'
+            condition: succeeded()
+            packageParentPath: '$(Build.SourcesDirectory)\artifacts\VSSetup\$(BuildConfiguration)\DevDivPackages'
+            packagesToPush: '$(Build.SourcesDirectory)\artifacts\VSSetup\$(BuildConfiguration)\DevDivPackages\**\*.nupkg'
+            allowPackageConflicts: true
+            nuGetFeedType: internal
+            publishVstsFeed: '97a41293-2972-4f48-8c0e-05493ae82010'
+
+          # Publish an artifact that the RoslynInsertionTool is able to find by its name.
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact VSSetup'
+            condition: succeeded()
+            targetPath: 'artifacts\VSSetup\$(BuildConfiguration)'
+            artifactName: 'VSSetup'
+
+          # Publish our NuPkgs as an artifact. The name of this artifact must be PackageArtifacts as the
+          # arcade templates depend on the name.
+          - output: buildArtifacts
+            displayName: 'Publish Artifact Packages'
+            condition: succeeded()
+            PathtoPublish: 'artifacts\packages\$(BuildConfiguration)'
+            ArtifactName: 'PackageArtifacts'
+
+          # Publish Asset Manifests for Build Asset Registry job
+          - output: buildArtifacts
+            displayName: 'Publish Asset Manifests'
+            condition: succeeded()
+            PathtoPublish: '$(Build.SourcesDirectory)/artifacts/log/$(BuildConfiguration)/AssetManifest'
+            ArtifactName: AssetManifests
+
+        steps:
+        - powershell: Write-Host "##vso[task.setvariable variable=OriginalBuildNumber;isreadonly=true]$('$(Build.BuildNumber)')"
+          displayName: Setting OriginalBuildNumber variable
+          condition: succeeded()
+
+        - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName;isreadonly=true]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
+          displayName: Setting SourceBranchName variable
+          condition: succeeded()
+
+        - task: Powershell@2
+          name: FancyBuild
+          displayName: Setting FancyBuild.BuildNumber
+          inputs:
+            targetType: inline
+            script: |
+              $pull_request = Invoke-RestMethod -Uri "https://api.github.com/repos/dotnet/roslyn/pulls/${{ parameters.PRNumber }}" `
+              -Headers @{
+                "Accept" = "application/vnd.github+json";
+                "X-GitHub-Api-Version" = "2022-11-28"
+              }
+              $buildNumberName = "$(OriginalBuildNumber) - $($pull_request.user.login) - '$($pull_request.title)'"
+              $buildNumberName = $buildNumberName -replace '["/:<>\|?@*]','_'
```

Looks like [the disallowed character set](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/run-number?view=azure-devops&tabs=yaml#how-large-can-a-run-number-be-and-what-characters-can-i-use) also includes single quote `'` and the whole number cannot end with a dot, so maybe that should be replaced as well? 
```suggestion
              $buildNumberName = $buildNumberName -replace '["/:<>\|?@*`'.]','_'
```

---

Thanks for pointing me to this. The error message did not include the single quote or information about the dot. I have updated so that we alway append " #" to the build number just like the Maestro Promotion pipeline does (now I know why!).

---

Hey wait a minute. Single quotes work fine. haha. As seen in the screenshot in the PR description.